dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -161,7 +161,24 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             THROW_HR_IF(E_INVALIDARG, taskClsid == GUID_NULL);
 
             auto registrationOptions = details.Options();
-            THROW_HR_IF(E_INVALIDARG, WI_AreAllFlagsClear(registrationOptions, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator));
+
+            auto isBackgroundTaskFlagSet{ WI_IsAnyFlagSet(registrationOptions, PushNotificationRegistrationOptions::PushTrigger | PushNotificationRegistrationOptions::ComActivator) };
+            THROW_HR_IF(E_INVALIDARG, isBackgroundTaskFlagSet && !IsActivatorSupported(registrationOptions));
+
+            auto isProtocolActivatorSet{ WI_IsFlagSet(registrationOptions, PushNotificationRegistrationOptions::ProtocolActivator) };
+            THROW_HR_IF(E_INVALIDARG, isProtocolActivatorSet && !IsActivatorSupported(registrationOptions));
+
+            if (isProtocolActivatorSet)
+            {
+                auto coInitialize = wil::CoInitializeEx();
```

Comment:
Why is this block still here? It can be moved to the block below right? 
if (WI_IsFlagSet(options, PushNotificationRegistrationOptions::ProtocolActivator))

Reply:
Basically the logic in this API should be simple if (protocol) { ... } else if (backgroundtrigger) ... else {}.

Right now the ordering and structure could use some work

Reply:
There is no such if in this function. Also, if we return at a single place then we have to return a null token (i.e. cookie is 0 and there is no registeredTask). It can improved after the whole feature is checked in and as part of token removal PR -> #1339.

Reply:
It should be: `if (Protocol) else { if(BI){} if(COM){} }`. We can't do an if-else on BI and then COM or vice versa.

Right now, returning a token is a problem for simplifying blocks if we are supposed to return in just one place. It can be addressed as part of the token removal PR -> #1339

Reply:
This actually sounds reasonable: if(Protocol){} else { if(BI){} if(COM){} }